sys/can/pkt.c

```diff
@@ -0,0 +1,287 @@
+/*
```

Comment:
Sure. For the record I will recap here why I need a memory pool for dynamic allocation.

First we need to allocate memory for incoming and outcoming CAN frames. This was the very first usage of the pkt module, and the first implementation was an array of frames. CAN frames are composed of at most 8 data bytes plus a CAN id.

Then I needed to allocate some memory for filter lists. They are used to make a link between incoming frames and user code, which can subscribe to some frames. Filters are also quite small, a filter is just a struct containing two `uint32_t`, with what needed around to route the frame to the right user.

The last usage is for isotp buffers. ISO-TP (transport protocol) is used to send more than 8 bytes on CAN. Up to 4095 bytes can theoritically be sent through an isotp channel. For this last use case, the allocated buffers can be bigger than in the two previous ones. Even if in the use cases I known it rarely goes above 500 bytes.